substrate/frame/support/src/traits/tokens/fungibles/lifetime.rs

```diff
@@ -32,6 +32,18 @@ pub trait Create<AccountId>: Inspect<AccountId> {
 	) -> DispatchResult;
 }
 
+/// Trait for refunding the deposit of a target asset account.
+pub trait Refund<AccountId> {
```

Comment:
Shouldn't this be `trait Refund<AccountId>: Inspect<AccountId>` to avoid defining `AssetId` and `Balance` again?

Reply:
I think it is more flexible, refund type do not have to implement the inspect trait.

Reply:
> Is there ever a usecase where `Refund` would be implemented on something other than the thing implementing `Inspect`.

Not likely in the context of the FRAME. I really see the point for super traits when some of the functions/methods used in a sub trait. But when need to decide convenience (no needs for type definition) vs flexibility (interface segregation), I will go for flexibility. 
In this case it's not that important, perhaps `Refund` wont need to be implemented on a type not implementing `Inspect`. I am fine with super trait.

Reply:
@georgepisaltu I had to set an additional bound to Currency type in Assets pallet for this. Please check and approve if you happy with it.
https://github.com/paritytech/polkadot-sdk/pull/3251/commits/290c0f1e84597374f6760e28aff731f96f2b4826

Reply:
however I do not see the benefits. in most cases you still have to define what Balance and AssetId are. and not you can not have a separate type implementing only refund.